core/templates/dev/head/pages/exploration-editor-page/services/changes-in-human-readable-form.service.ts

```diff
@@ -16,283 +16,306 @@
+    if (Array.isArray(newValue) && Array.isArray(oldValue)) {
+      result = (newValue.length > oldValue.length) ?
           'added' : (newValue.length === oldValue.length) ?
-            'edited' : 'deleted';
-      } else {
-        if (!UtilsService.isEmpty(oldValue)) {
-          if (!UtilsService.isEmpty(newValue)) {
-            result = 'edited';
-          } else {
-            result = 'deleted';
-          }
-        } else if (!UtilsService.isEmpty(newValue)) {
-          result = 'added';
+              'edited' : 'deleted';
+    } else {
+      if (!this.utilsService.isEmpty(oldValue)) {
+        if (!this.utilsService.isEmpty(newValue)) {
+          result = 'edited';
+        } else {
+          result = 'deleted';
         }
+      } else if (!this.utilsService.isEmpty(newValue)) {
+        result = 'added';
       }
-      return result;
-    };
-
-    var makeHumanReadable = function(lostChanges) {
-      var outerHtml = angular.element('<ul></ul>');
-      var stateWiseEditsMapping = {};
-      // The variable stateWiseEditsMapping stores the edits grouped by state.
-      // For instance, you made the following edits:
-      // 1. Changed content to 'Welcome!' instead of '' in 'Introduction'.
-      // 2. Added an interaction in this state.
-      // 2. Added a new state 'End'.
-      // 3. Ended Exporation from state 'End'.
-      // stateWiseEditsMapping will look something like this:
-      // - 'Introduction': [
-      //   - 'Edited Content: Welcome!',:
-      //   - 'Added Interaction: Continue',
-      //   - 'Added interaction customizations']
-      // - 'End': ['Ended exploration']
+    }
+    return result;
+  }
+  // TODO(#7176): Replace 'any' with the exact type.
+  _makeHumanReadable(lostChanges: Array<any>): HTMLElement {
+    let outerHtml = this.document.createElement('<ul></ul>');
+    let stateWiseEditsMapping = {};
+    // The letiable stateWiseEditsMapping stores the edits grouped by state.
+    // For instance, you made the following edits:
+    // 1. Changed content to 'Welcome!' instead of '' in 'Introduction'.
+    // 2. Added an interaction in this state.
+    // 2. Added a new state 'End'.
+    // 3. Ended Exporation from state 'End'.
+    // stateWiseEditsMapping will look something like this:
+    // - 'Introduction': [
+    //   - 'Edited Content: Welcome!',:
+    //   - 'Added Interaction: Continue',
+    //   - 'Added interaction customizations']
+    // - 'End': ['Ended exploration']
 
-      lostChanges.forEach(function(lostChange) {
-        switch (lostChange.cmd) {
-          case CMD_ADD_STATE:
-            outerHtml.append(
-              angular.element('<li></li>').html(
-                'Added state: ' + lostChange.state_name));
-            break;
-          case CMD_RENAME_STATE:
-            outerHtml.append(
-              angular.element('<li></li>').html(
-                'Renamed state: ' + lostChange.old_state_name + ' to ' +
+    lostChanges.forEach((lostChange) => {
```

Comment:
Type `lostChange` or put the any comment.

Reply:
Done